src/main/webapp/app/exercises/shared/team/teams-import-dialog/teams-import-dialog.component.ts

```diff
@@ -239,11 +287,56 @@ export class TeamsImportDialogComponent implements OnInit, OnDestroy {
         if (this.isSubmitDisabled) {
             return;
         }
-        this.isImporting = true;
-        this.teamService.importTeamsFromSourceExercise(this.exercise, this.sourceExercise, this.importStrategy!).subscribe(
-            (res) => this.onSaveSuccess(res),
-            () => this.onSaveError(),
-        );
+        if (this.showImportFromExercise) {
+            this.isImporting = true;
+            this.teamService.importTeamsFromSourceExercise(this.exercise, this.sourceExercise!, this.importStrategy!).subscribe(
+                (res) => this.onSaveSuccess(res),
+                (error) => this.onSaveError(error),
+            );
+        } else if (this.sourceTeams) {
+            this.notFoundRegistrationNumbers = [];
+            this.teamService.importTeams(this.exercise, this.sourceTeams, this.importStrategy!).subscribe(
+                (res) => this.onSaveSuccess(res),
+                (error) => this.onSaveError(error),
+            );
+        }
+    }
+
+    /**
+     * Update source teams to given teams
+     * @param {Team[]} fileTeams - Teams which its students only have visible registration number
+     */
+    onTeamsChanged(fileTeams: Team[]) {
```

Comment:
this method is rather complex, please split it into multiple smaller methods and make sure to find good method names and add more documentation